### DISCUSSION TITLE
QEMU+EDK2 cannot passthrough an iso as cdrom to VM for boot option

### DISCUSSION THREAD

**Discussion by enzoyes** [tianocore/edk2#4753] [Q&A]
Enviroment:
1. edk2-aarch64-20210527gite1999b264f1f-3.el8.noarch
2. qemu 4.2.1
3. libvirt 6.2.0
4. aarch64

xml description slice as below
`  
 <disk type='file' device='cdrom'>
      <driver name='qemu' type='raw' cache='none' io='native'/>
      <source file='$path_of_so'/>
      <target dev='vde' bus='sata'/>
      <readonly/>
    </disk>
`
i want to use cdrom as boot option, however, boot stop at UEFI shell and it seems no efi is detected(no BLK and no FS0).

and also, i made a experiment, pass the iso to a system directly and i can see the cdrom in the system. So, IMHO, it means qemu could get the device, and maybe something went wrong for UEFI find the cdrom.

i've made some change, like change sata to ide(not supported in the qemu version).

Very confused about the problem, anyone know the reason? Kind for your discussions and help.

**Comment by zh-jq** [comment]
@enzoyes you can change the bus type to 'usb'.
```xml
<target dev='sda' bus='usb'/>
```
